spec.md

```diff
@@ -4134,591 +4046,641 @@ The error codes that MAY be included in the response body are enumerated below:
-|-----------------------|------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
-| `DIGEST_INVALID`      | provided digest did not match uploaded content | When a blob is uploaded, the registry will check that the content matches the digest provided by the client.The error MAY include a detail structure with the key "digest", including the invalid digest string.This error MAY also be returned when a manifest includes an invalid layer digest. |
-| `NAME_INVALID`        | invalid repository name                        | Invalid repository name encountered either during manifest validation or any API operation.                                                                                                                                                                                                       |
-| `BLOB_UPLOAD_INVALID` | blob upload invalid                            | The blob upload encountered an error and can no longer proceed.                                                                                                                                                                                                                                   |
-| `UNSUPPORTED`         | The operation is unsupported.                  | The operation was unsupported due to a missing implementation or invalid set of parameters.                                                                                                                                                                                                       |
+- **Registry**: a HTTP service which implements this spec
+- **Client**: a tool that communicates with registries over HTTP
+- **Push**: the act of uploading content to a registry
+- **Pull**: the act of downloading content from a registry
+- **Artifact**: a single piece of content, made up of a manifest and one or more layers
+- **Manifest**: a JSON document which defines an artifact
+- **Layer**: a single part of all the parts which comprise an artifact
+- **Config**: a special layer defined at the top of a manifest containing artifact metadata
+- **Blob**: a single binary content stored in a registry
+- **Digest**: a unique blob identifier
+- **Content**: a general term for content that can be downloaded from a registry (manifest or blob)
 
-###### On Failure: Not Found
+### Document Language
```

Comment:
Did you want to move this to the end of the document? RFCs typically put this language at the very beginning in a "Terminology" section.